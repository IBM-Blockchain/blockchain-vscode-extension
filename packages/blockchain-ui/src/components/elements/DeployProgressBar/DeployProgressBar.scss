#root {

    .deploy-page-container{
        .bx--tab-content {
            width: 100%;
            padding-left: 0px;
            background-color: transparent;
        }
    
        .bx--progress-line {
            width: 100%;
        }
    
        .bx--progress-step {
            width: 100%;
        }
    }
}
