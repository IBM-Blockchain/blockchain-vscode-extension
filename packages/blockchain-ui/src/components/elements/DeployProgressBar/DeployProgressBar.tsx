import React, { Component } from 'react';
import { ProgressIndicator, ProgressStep } from 'carbon-components-react';
import './DeployProgressBar.scss';

interface IProps {
    currentIndex: number;
}

class DeployProgressBar extends Component<IProps> {

    // For the deploy progress bar

    render(): JSX.Element {
        return (
            <div className='bx--col-lg-10'>
                <ProgressIndicator currentIndex={this.props.currentIndex} vertical={false}>

                    <ProgressStep
                        label='Step 1'
                        secondaryLabel='Choose smart contract'
                        description=''

                    />
                    <ProgressStep
                        label='Step 2'
                        secondaryLabel='Create definition'
                        description=''

                    />
                    <ProgressStep
                        label='Step 3'
                        secondaryLabel='Deploy'
                        description=''
                    />

                </ProgressIndicator>
            </div>
        );
    }
}

export default DeployProgressBar;
