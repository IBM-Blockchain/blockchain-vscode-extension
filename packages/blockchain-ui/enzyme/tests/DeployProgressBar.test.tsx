import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import chai from 'chai';
import sinonChai from 'sinon-chai';

import DeployProgressBar from '../../src/components/elements/DeployProgressBar/DeployProgressBar';

chai.should();
chai.use(sinonChai);

describe('DeployProgressBar component', () => {
    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<DeployProgressBar currentIndex={0}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
