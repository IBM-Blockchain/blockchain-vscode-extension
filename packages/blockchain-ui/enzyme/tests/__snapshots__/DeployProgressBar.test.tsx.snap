// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeployProgressBar component should render the expected snapshot 1`] = `
<div
  className="bx--col-lg-10"
>
  <ul
    className="bx--progress"
  >
    <li
      className="bx--progress-step bx--progress-step--current"
    >
      <button
        className="bx--progress-step-button bx--progress-step-button--unclickable"
        onKeyDown={[Function]}
        tabIndex={-1}
      >
        <span
          className="bx--assistive-text"
        >
          Current
        </span>
        <svg
          aria-hidden={true}
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="16"
            cy="16"
            r="10"
          />
          <path
            d="M16,30A14,14,0,1,1,30,16,14.0158,14.0158,0,0,1,16,30ZM16,4A12,12,0,1,0,28,16,12.0137,12.0137,0,0,0,16,4Z"
          />
          <title>
            
          </title>
        </svg>
        <p
          className="bx--progress-label"
        >
          Step 1
        </p>
        <p
          className="bx--progress-optional"
        >
          Choose smart contract
        </p>
        <span
          className="bx--progress-line"
        />
      </button>
    </li>
    <li
      className="bx--progress-step bx--progress-step--incomplete"
    >
      <button
        className="bx--progress-step-button bx--progress-step-button--unclickable"
        onKeyDown={[Function]}
        tabIndex={-1}
      >
        <span
          className="bx--assistive-text"
        >
          Incomplete
        </span>
        <svg
          aria-hidden={true}
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16,2A14,14,0,1,0,30,16,14,14,0,0,0,16,2Zm0,26A12,12,0,1,1,28,16,12,12,0,0,1,16,28Z"
          />
          <title>
            
          </title>
        </svg>
        <p
          className="bx--progress-label"
        >
          Step 2
        </p>
        <p
          className="bx--progress-optional"
        >
          Create definition
        </p>
        <span
          className="bx--progress-line"
        />
      </button>
    </li>
    <li
      className="bx--progress-step bx--progress-step--incomplete"
    >
      <button
        className="bx--progress-step-button bx--progress-step-button--unclickable"
        onKeyDown={[Function]}
        tabIndex={-1}
      >
        <span
          className="bx--assistive-text"
        >
          Incomplete
        </span>
        <svg
          aria-hidden={true}
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16,2A14,14,0,1,0,30,16,14,14,0,0,0,16,2Zm0,26A12,12,0,1,1,28,16,12,12,0,0,1,16,28Z"
          />
          <title>
            
          </title>
        </svg>
        <p
          className="bx--progress-label"
        >
          Step 3
        </p>
        <p
          className="bx--progress-optional"
        >
          Deploy
        </p>
        <span
          className="bx--progress-line"
        />
      </button>
    </li>
  </ul>
</div>
`;
